lighthouse-core/lib/lh-error.js

```diff
@@ -70,116 +105,120 @@ const ERRORS = {
   // Screenshot/speedline errors
   NO_SPEEDLINE_FRAMES: {
     code: 'NO_SPEEDLINE_FRAMES',
-    message: strings.didntCollectScreenshots,
+    message: UIStrings.didntCollectScreenshots,
     lhrRuntimeError: true,
   },
   SPEEDINDEX_OF_ZERO: {
     code: 'SPEEDINDEX_OF_ZERO',
-    message: strings.didntCollectScreenshots,
+    message: UIStrings.didntCollectScreenshots,
     lhrRuntimeError: true,
   },
   NO_SCREENSHOTS: {
     code: 'NO_SCREENSHOTS',
-    message: strings.didntCollectScreenshots,
+    message: UIStrings.didntCollectScreenshots,
     lhrRuntimeError: true,
   },
   INVALID_SPEEDLINE: {
     code: 'INVALID_SPEEDLINE',
-    message: strings.didntCollectScreenshots,
+    message: UIStrings.didntCollectScreenshots,
     lhrRuntimeError: true,
   },
 
   // Trace parsing errors
   NO_TRACING_STARTED: {
     code: 'NO_TRACING_STARTED',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_NAVSTART: {
     code: 'NO_NAVSTART',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_FCP: {
     code: 'NO_FCP',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_DCL: {
     code: 'NO_DCL',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_FMP: {
     code: 'NO_FMP',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
   },
 
   // TTI calculation failures
-  FMP_TOO_LATE_FOR_FCPUI: {code: 'FMP_TOO_LATE_FOR_FCPUI', message: strings.pageLoadTookTooLong},
-  NO_FCPUI_IDLE_PERIOD: {code: 'NO_FCPUI_IDLE_PERIOD', message: strings.pageLoadTookTooLong},
-  NO_TTI_CPU_IDLE_PERIOD: {code: 'NO_TTI_CPU_IDLE_PERIOD', message: strings.pageLoadTookTooLong},
+  FMP_TOO_LATE_FOR_FCPUI: {code: 'FMP_TOO_LATE_FOR_FCPUI', message: UIStrings.pageLoadTookTooLong},
+  NO_FCPUI_IDLE_PERIOD: {code: 'NO_FCPUI_IDLE_PERIOD', message: UIStrings.pageLoadTookTooLong},
+  NO_TTI_CPU_IDLE_PERIOD: {code: 'NO_TTI_CPU_IDLE_PERIOD', message: UIStrings.pageLoadTookTooLong},
   NO_TTI_NETWORK_IDLE_PERIOD: {
     code: 'NO_TTI_NETWORK_IDLE_PERIOD',
-    message: strings.pageLoadTookTooLong,
+    message: UIStrings.pageLoadTookTooLong,
   },
 
   // Page load failures
   NO_DOCUMENT_REQUEST: {
     code: 'NO_DOCUMENT_REQUEST',
-    message: strings.pageLoadFailed,
+    message: UIStrings.pageLoadFailed,
     lhrRuntimeError: true,
   },
   /* Used when DevTools reports loading failed. Usually an internal (Chrome) issue. */
   FAILED_DOCUMENT_REQUEST: {
     code: 'FAILED_DOCUMENT_REQUEST',
-    message: strings.pageLoadFailed,
+    message: UIStrings.pageLoadFailedWithDetails,
     lhrRuntimeError: true,
+    errorDetails: 'No Failure Description Loaded.',
```

Comment:
do we want defaults for these? What happens if you don't provide them? Throwing if you don't provide one seems ideal

Reply:
Yeah the ICU message translator throws.  With a 'value that wasn't replaced', good point, we probably do want to throw in order to avoid a silent lack of real replacement.